File: JobManagementSystem.Core/Models/Job.cs
using System;

namespace JobManagementSystem.Core.Models
{
    public class Job
    {
        public Guid Id { get; set; }
        public required string Name { get; set; }
        public JobPriority Priority { get; set; }
        public JobStatus Status { get; set; }
        public DateTime StartTime { get; set; }
        public DateTime? EndTime { get; set; }
        public int Progress { get; set; }
        public DateTime CreatedAt { get; set; }
        public string? ErrorMessage { get; set; }
        public int RetryCount { get; set; }
    }

    public enum JobPriority
    {
        Regular,
        High
    }

    public enum JobStatus
    {
        Pending,
        Running,
        Completed,
        Failed,
        Stopped
    }
} 

File: JobManagementSystem.Web/src/types/Job.ts
export enum JobPriority {
  Regular = 0,
  High = 1
}

export enum JobStatus {
  Pending = 'Pending',
  Running = 'Running',
  Completed = 'Completed',
  Failed = 'Failed',
  Stopped = 'Stopped'
}

export interface Job {
  id: string;  // Always store as string internally
  name: string;
  priority: JobPriority;
  status: JobStatus;
  progress: number;
  createdAt: string;
  startedAt?: string;
  completedAt?: string;
  errorMessage?: string;
  workerNodeId?: string;
  lastClaimTime?: string;
} 